buildchain/buildchain/targets/local_image.py

```diff
@@ -62,10 +117,56 @@ def __init__(
             **kwargs
         )
 
-
     dockerfile   = property(operator.attrgetter('_dockerfile'))
     save_on_disk = property(operator.attrgetter('_save'))
     build_args   = property(operator.attrgetter('_build_args'))
+    dep_re = re.compile(
+        r'^\s*(COPY|ADD)( --[^ ]+)* (?P<src>[^ ]+) (?P<dst>[^ ]+)\s*$'
+    )
+
+    def load_deps_from_dockerfile(self) -> List[DockerFileDep]:
+        """Compute file dependencies from Dockerfile."""
+        dep_keywords = ('COPY', 'ADD')
+
+        with self.dockerfile.open('r', encoding='utf8') as dockerfile:
+            docker_lines = dockerfile.readlines()
+
+        dep_lines = [
+            (pos, line.strip(), self.dep_re.match(line.strip()))
+            for (pos, line) in enumerate(docker_lines, 1)
+            if line.strip().startswith(dep_keywords)
+        ]
+        deps = []
+        for (pos, line, match) in dep_lines:
+            if match is None:
+                continue
+            deps.append(
+                DockerFileDep(
+                    line_no=pos,
+                    line_contents=line,
+                    line_match=match,
+                    dockerfile_path=self.dockerfile
+                )
+            )
+        return deps
+
+    def check_dockerfile_dependencies(self) -> Union[TaskError, bool]:
+        """Verify task file dependencies against computed file dependencies."""
+        error_tplt = (
+            'Missing Dockerfile file dependencies in'
+            ' build target: \n{dockerfile}:\n\t{errors}'
+        )
+        deps = self.load_deps_from_dockerfile()
+        errors: List[str] = []
+        for dep in deps:
+            errors.extend(dep.verify(self.file_dep))
+        if errors:
+            error_message = error_tplt.format(
+                dockerfile=self.dockerfile,
+                errors='\n\t'.join(errors)
+            )
+            return TaskError(msg=error_message)
+        return True
```

Comment:
`return None` would be more consistent (since you don't return `False` in the other case).

Reply:
Yup, same return convention as the Docker API PR. Done.